src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsMenuBarUI.java

```diff
@@ -148,10 +149,15 @@ public void actionPerformed(ActionEvent e) {
                 MenuElement[] path = new MenuElement[2];
                 path[0] = (MenuElement)menuBar;
                 path[1] = (MenuElement)menu;
-                msm.setSelectedPath(path);
-
-                // show mnemonics
-                WindowsLookAndFeel.setMnemonicHidden(false);
+                if (!mnemonicShowHideFlag) {
```

Comment:
I don't think you need a field at all.

The entire method should look like this:

```java
        public void actionPerformed(ActionEvent e) {
            JMenuBar menuBar = (JMenuBar)e.getSource();
            JMenu menu = menuBar.getMenu(0);
            if (menu != null) {
                MenuSelectionManager msm =
                    MenuSelectionManager.defaultManager();
                MenuElement[] selectedPath = msm.getSelectedPath();
                if (selectedPath.length > 0 && (selectedPath[0] instanceof JMenuBar)) {
                    msm.clearSelectedPath();

                    WindowsLookAndFeel.setMnemonicHidden(false);
                } else {
                    MenuElement[] path = {menuBar, menu};
                    msm.setSelectedPath(path);

                    WindowsLookAndFeel.setMnemonicHidden(false);
                }
                WindowsLookAndFeel.repaintRootPane(menuBar);
            }
        }
```

If there's a selection path and it starts with the menu bar, clear selection and hide mnemonics.

Otherwise, select the menu bar and its first menu.